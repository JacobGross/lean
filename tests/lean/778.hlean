open bool
definition is_equiv_bnot : is_equiv bnot :=
begin
  fapply is_equiv.mk,
    exact bnot,
    all_goals (intro b;cases b),
    state,
    repeat reflexivity
end
