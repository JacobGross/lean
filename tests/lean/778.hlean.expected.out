778.hlean:7:4: proof state

⊢ bnot (bnot ff) = ff


⊢ bnot (bnot tt) = tt


⊢ bnot (bnot ff) = ff


⊢ bnot (bnot tt) = tt


⊢ bool.cases_on (bnot ff) ?M_1 ?M_2 = eq.ap bnot (bool.cases_on ff ?M_3 ?M_4)


⊢ bool.cases_on (bnot tt) ?M_1 ?M_2 = eq.ap bnot (bool.cases_on tt ?M_3 ?M_4)
